app/microsoft/endpoints/folder.py

```diff
@@ -13,58 +12,66 @@ class FolderEndpoint(BaseEndpoint):
-        Get the Folder inside the Group (Staging or Production) Drive (filesystem).
+        Get the Folder inside the Group Drive (filesystem).
         Returns driveItem object or None.
         """
-        url = os.path.join(self.base_url, f"root:/{path}")
+        url = os.path.join(self.MIDDLE_URL, path)
+
         return super().get(url)
 
     def get_children(self, path):
         """
-        Get the child items inside a folder.
+        Get child items (folders, files) inside current folder
         """
-        url = os.path.join(self.base_url, f"root:/{path}:/children")
-        data = super().get(url)
-        return data
+        url = os.path.join(self.MIDDLE_URL, f"{path}:/children")
+
+        return super().get(url)
 
     def get_all_files(self, path):
         """
-        Get all files inside a folder.
+        Recursively get all files inside current folder.
         """
-        url = os.path.join(self.base_url, f"root:/{path}:/children")
-        children_data = super().get(url)
-        files = []
+        children_data = self.get_children(path)
         children = children_data["value"] if children_data else []
+
+        all_files = []
+
         for item in children:
             if item.get("file"):
-                # It's a file
-                files.append(item)
+                all_files.append(item)
             elif item.get("folder"):
-                # It's a folder
                 has_children = item.get("folder", {}).get("childCount", 0) > 0
+
                 if has_children:
-                    folder_path = os.path.join(path, item["name"])
-                    folder_child_files = self.get_all_files(folder_path)
-                    files += folder_child_files
+                    folder_children = self.get_all_files(
+                        os.path.join(path, item["name"])
+                    )
+                    all_files += folder_children
 
-        return files
+        return all_files
 
-    def download_file(self, file_drive_id):
+    def download_file(self, file_id):
         """
-        Returns filename, minemtype, file bytes
+        Returns file name, MIME type, file bytes
         """
-        file_url = os.path.join(self.base_url, f"items/{file_drive_id}")
-        file_data = super().get(file_url)
-        filename = file_data["name"]
+        file_data = super().get(
+            f"groups/{settings.MS_GRAPH_GROUP_ID}/drive/items/{file_id}"
+        )
+        file_name = file_data["name"]
         mimetype = file_data["file"]["mimeType"]
-        url = os.path.join(BASE_URL, self.base_url, f"items/{file_drive_id}/content")
+
+        url = os.path.join(
+            BASE_URL,
+            f"groups/{settings.MS_GRAPH_GROUP_ID}/drive/items/{file_id}/content",
+        )
         resp = requests.get(url, headers=self.headers, stream=False)
         resp.raise_for_status()
-        return filename, mimetype, resp.content
+
+        return file_name, mimetype, resp.content
 
```

Comment:
I don't see the point of having whitespace before the return statement, but it's not a big deal either way
note that the codebase sometimes formatted with Black so try not to lean too heavily on idiosyncratic formatting to convey meaning